src/main/java/net/openhft/chronicle/network/FileBasedHostnamePortLookupTable.java

```diff
@@ -0,0 +1,204 @@
+package net.openhft.chronicle.network;
+
+import net.openhft.chronicle.bytes.MappedBytes;
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.OS;
+import net.openhft.chronicle.core.StackTrace;
+import net.openhft.chronicle.core.io.Closeable;
+import net.openhft.chronicle.core.io.IORuntimeException;
+import net.openhft.chronicle.wire.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static net.openhft.chronicle.core.util.Time.sleep;
+
+/**
+ * Stores the mappings in a shared file, will work across processes
+ */
+public class FileBasedHostnamePortLookupTable implements HostnamePortLookupTable, java.io.Closeable {
+
+    private static final long LOCK_TIMEOUT_MS = 4_000;
+    private static final int DELETE_TABLE_FILE_TIMEOUT_MS = 1_000;
+    private static final int PID = Jvm.getProcessId();
+    private static final String DEFAULT_FILE_NAME = "shared_hostname_mappings";
```

Comment:
maybe put "chronicle" in the name.
Should be in temp folder?

Reply:
Yeah tmp is probably a good place, it just needs to be consistent across processes.

Reply:
Given we are test-scoped, maybe not necessary to be in tmp